src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10100,49 +10116,122 @@ private bool ScanCast(bool forPattern = false)
             return (type == ScanTypeFlags.GenericTypeOrMethod || type == ScanTypeFlags.GenericTypeOrExpression || type == ScanTypeFlags.NonGenericTypeOrExpression || type == ScanTypeFlags.TupleType) && CanFollowCast(this.CurrentToken.Kind);
         }
 
-        private bool ScanAsyncLambda(Precedence precedence)
+        private bool IsPossibleLambdaExpression(Precedence precedence)
         {
-            // Adapted from CParser::ScanAsyncLambda
-
-            // Precedence must not exceed that of lambdas
+            // Only call into this if after `static` or after a legal identifier.
+            Debug.Assert(
+                this.CurrentToken.Kind == SyntaxKind.StaticKeyword ||
+                this.IsTrueIdentifier(this.CurrentToken));
             if (precedence > Precedence.Lambda)
             {
                 return false;
             }
 
-            // Async lambda must start with 'async'
-            if (this.CurrentToken.ContextualKind != SyntaxKind.AsyncKeyword)
+            // If we start with `static` or `async static` then just jump past those and do the
+            // analysis after that point.  Note, we don't just past `async` in `static async`
+            // because that `async` may not be a modifier (it may just be an identifier) and we have
+            // to figure out what it is.
+
+            var peekIndex = 0;
+            var seenStatic = false;
+            if (this.CurrentToken.Kind == SyntaxKind.StaticKeyword)
+            {
+                peekIndex++;
```

Comment:
>peekIndex++ [](start = 16, length = 11)

Consider setting the value explicitly, here and in `else if`. Or set both locals explicitly in each branch:
```C#
int peekIndex;
bool seenStatic;
if (StaticKeyword)
{
    peekIndex = 1;
    seenStatic = true;
}
else if (AsyncKeyword)
{
    peekIndex = 2;
    seenStatic = true;
}
else
{
    peekIndex = 0;
    seenStatic = false;
}
```

Reply:
Done.  Thanks!